src/client/entity-editor/common/validation-label.js

```diff
@@ -85,10 +85,11 @@ function ValidationLabel({
 	empty,
 	error,
 	errorMessage,
+	warnMessage,
 	warn
 }: Props) {
-	const warnElement = (warn && !empty && !error) &&
-		<span className={contextualColor(empty, error, warn)}> Are you sure? You entered a date in the future!</span>;
+	const warnElement = (warn && warnMessage && !empty && !error) &&
```

Comment:
There is no need to check for the presence of `warnMessage` here, for two reasons:
If no `warnMessage` prop is passed to the component, it will default to an empty string (see my other comment).
Also, the warning is used in most cases without a warningMessage, so we want to make it works without it. (now I am wondering if that should be the same for `errorMessage &&` on the line below. I will investigate and possibly open another issue.)